server/src/test/scala/services/graphstages/ArrivalsGraphStagePaxNosSpec.scala

```diff
@@ -0,0 +1,202 @@
+    forecastSource.offer(List(arrival(actPax = Option(100), scheduledMillis = nowForThisTest.millisSinceEpoch)))
+    liveSource.offer(List(arrival(actPax = Option(0), scheduledMillis = nowForThisTest.millisSinceEpoch)))
+
+    probe.fishForMessage(2 seconds) {
+      case ArrivalsDiff(toUpdate, _) =>
+        toUpdate.exists {
+          case (_, a) => a.ActPax == Option(0)
+        }
+    }
+    success
+  }
+
+  "Given a live arrival with 0 pax in the live feed that is scheduled for now " +
+    "Then we should assume that 0 is the correct number of pax" >> {
+    val probe = TestProbe("arrivals")
+    val (aclSource, _, _, liveSource) = TestableArrivalsGraphStage(probe, buildArrivalsGraphStage).run
+
+    aclSource.offer(List(arrival(actPax = Option(189), scheduledMillis = nowForThisTest.millisSinceEpoch)))
+    liveSource.offer(List(arrival(actPax = Option(0), scheduledMillis = nowForThisTest.millisSinceEpoch)))
+
+    probe.fishForMessage(2 seconds) {
+      case ArrivalsDiff(toUpdate, _) =>
+        toUpdate.exists {
+          case (_, a) => a.ActPax == Option(0)
+        }
+    }
+    success
+  }
+
+  "Given a live arrival with 0 pax in the live feed that has landed regardless of schedule time " +
+    "Then we should assume that 0 is the correct number of pax" >> {
+    val probe = TestProbe("arrivals")
+    val (aclSource, _, _, liveSource) = TestableArrivalsGraphStage(probe, buildArrivalsGraphStage).run
+
+    aclSource.offer(
+      List(arrival(actPax = Option(189), scheduledMillis = nowForThisTest.addHours(8).millisSinceEpoch))
+    )
+    liveSource.offer(
+      List(arrival(
+        actPax = Option(0),
+        scheduledMillis = nowForThisTest.addHours(8).millisSinceEpoch,
+        actual = Option(nowForThisTest.millisSinceEpoch)
+      ))
+    )
+
+    probe.fishForMessage(2 seconds) {
+      case ArrivalsDiff(toUpdate, _) =>
+        toUpdate.exists {
+          case (_, a) => a.ActPax == Option(0)
+        }
+    }
+    success
+  }
+
+  "Given a live arrival with 0 pax in the live feed that has an ActualChox regardless of schedule time " +
+    "Then we should assume that 0 is the correct number of pax" >> {
+    val probe = TestProbe("arrivals")
+    val (aclSource, _, _, liveSource) = TestableArrivalsGraphStage(probe, buildArrivalsGraphStage).run
+
+    aclSource.offer(
+      List(arrival(actPax = Option(189), scheduledMillis = nowForThisTest.addHours(8).millisSinceEpoch))
+    )
+    liveSource.offer(
+      List(arrival(
+        actPax = Option(0),
+        scheduledMillis = nowForThisTest.addHours(8).millisSinceEpoch,
+        actChox = Option(nowForThisTest.millisSinceEpoch)
+      ))
+    )
+
+    probe.fishForMessage(2 seconds) {
+      case ArrivalsDiff(toUpdate, _) =>
+        toUpdate.exists {
+          case (_, a) => a.ActPax == Option(0)
+        }
+    }
+    success
+  }
+
+  def arrival(estimated: Option[Long] = None,
```

Comment:
We've already for an `ArrivalGenerator` object that creates arrivals based off only the fields you want to set, so we don't need this addition one